cpp/velox/memory/VeloxMemoryManager.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "memory/MemoryAllocator.h"
+#include "memory/MemoryManager.h"
+#include "velox/common/memory/Memory.h"
+#include "velox/common/memory/MemoryPool.h"
+
+namespace gluten {
+
+class VeloxMemoryManager final : public MemoryManager {
+ public:
+  explicit VeloxMemoryManager(
+      std::string name,
+      std::shared_ptr<MemoryAllocator> allocator,
+      std::shared_ptr<AllocationListener> listener);
+
+  ~VeloxMemoryManager() {
+    // Release order is important, from leaf to root.
+    veloxLeafPool_ = nullptr;
+    veloxPool_ = nullptr;
+    veloxMemoryManager_ = nullptr;
+  }
```

Comment:
Let's remove the dtor

Reply:
I'm not sure the release order is correct by default, do you?